server/app/com/xsn/explorer/services/StatisticsService.scala

```diff
@@ -19,27 +19,64 @@ class StatisticsService @Inject()(
     xsnService: XSNService,
     actorSystem: ActorSystem,
     statisticsFutureDataHandler: StatisticsFutureDataHandler,
-    merchantnodeRepository: MerchantnodeRepository
+    merchantnodeRepository: MerchantnodeRepository,
+    masternodeRepository: MasternodeRepository,
+    nodeStatsRepository: NodeStatsRepository
 )(
     implicit ec: ExecutionContext
 ) {
 
   def getStatistics(): FutureApplicationResult[StatisticsDetails] = {
     val dbStats = statisticsFutureDataHandler.getStatistics()
-    val mnStats = xsnService.getMasternodeCount()
+    val mnStats = masternodeRepository.getCount()
     val tposStats = merchantnodeRepository.getCount()
     val difficultyF = xsnService.getDifficulty()
 
     val result = for {
       stats <- dbStats.toFutureOr
-      mnCount <- discardErrors(mnStats).toFutureOr
+      mnCount <- mnStats.map(x => Good(Some(x))).toFutureOr
       difficulty <- discardErrors(difficultyF).toFutureOr
       tposCount <- tposStats.map(x => Good(Some(x))).toFutureOr
     } yield StatisticsDetails(stats, mnCount, tposCount, difficulty)
 
     result.toFuture
   }
 
+  def getNodeStatistics(): FutureApplicationResult[NodeStatistics] = {
+    val mnCount = masternodeRepository.getCount()
+    val mnEnabledCount = masternodeRepository.getEnabledCount()
+    val mnProtocols = masternodeRepository.getProtocols()
+    val tposCount = merchantnodeRepository.getCount()
+    val tposEnabledCount = merchantnodeRepository.getEnabledCount()
+    val tposProtocols = merchantnodeRepository.getProtocols()
+    val coinsStaking = nodeStatsRepository.getCoinsStaking()
+
+    val result = for {
+      mnCountVal <- mnCount.map(x => Good(x)).toFutureOr
+      mnEnabledCountVal <- mnEnabledCount.map(x => Good(x)).toFutureOr
+      mnProtocolsVal <- mnProtocols.map(x => Good(x)).toFutureOr
+      tposCountVal <- tposCount.map(x => Good(x)).toFutureOr
+      tposEnabledCountVal <- tposEnabledCount.map(x => Good(x)).toFutureOr
+      tposProtocolsVal <- tposProtocols.map(x => Good(x)).toFutureOr
+      coinsStakingVal <- coinsStaking.map(x => Good(x)).toFutureOr
+    } yield NodeStatistics(mnCountVal, mnEnabledCountVal, mnProtocolsVal, tposCountVal, tposEnabledCountVal, tposProtocolsVal, coinsStakingVal)
+
+    result.toFuture
+  }
+
+  def getCoinsStaking(): FutureApplicationResult[BigDecimal] = {
+    val tposNodes = merchantnodeRepository.getAll()
+
+    val result = for {
+      tposNodesList <- tposNodes.map(x => Good(x)).toFutureOr
+
+      coinsStaking <- tposNodesList.map(t => t.payee.string).map(statisticsFutureDataHandler.getCoinsInAddress).toFutureOr
```

Comment:
You need to verify this because I believe the payee is the merchant addess, which receives rewards, instead of what's actually staking